package_scripts/start.ts

```diff
@@ -0,0 +1,252 @@
+import * as path from "node:path"
+
+import { program } from "commander"
+import concurrently from "concurrently"
+import inquirer from "inquirer"
+import * as color from "ansi-colors"
+import execSh from "exec-sh"
+
+import { env, envSchema } from "../env"
+import * as u from "./util"
+
+type HugoServerOptions = {
+  port: number
+  bind: string
+  config: string
+  themesDir: string
+  renderToDisk: boolean
+}
+
+const hugoServer = (
+  opts: Partial<HugoServerOptions> & Pick<HugoServerOptions, "config">
+) => {
+  const allOpts: HugoServerOptions = {
+    port:         3000,
+    bind:         "0.0.0.0",
+    themesDir:    process.cwd(),
+    renderToDisk: true,
+    ...opts
+  }
+  return `hugo server ${u.getOptions(allOpts)}`
+}
+
+/**
+ * Start webpack dev server and another script, specified by `other`.
+ */
+const startWebpackAnd = ({
+  command,
+  name,
+  cwd
+}: {
+  command: string
+  name: string
+  cwd?: string
+}): void =>
+  concurrently(
+    [
+      {
+        command:     "yarn start:webpack",
+        name:        "webpack",
+        prefixColor: "blue"
+      },
+      {
+        command,
+        name,
+        cwd,
+        env:         { ...process.env, ...env },
+        prefixColor: "red"
+      }
+    ],
+    {
+      killOthers: ["failure", "success"]
+    }
+  )
+
+/**
+ * Check whether `dirpath` is a non-empty directory. If so, return. If not,
+ * clone `repoUrl` into `dirpath` after prompting user for confirmation.
+ */
+const ensureGitContent = async ({
+  dirpath,
+  repoUrl,
+  explainMissingContent
+}: {
+  dirpath: string
+  repoUrl: string
+  explainMissingContent: string
+}) => {
+  if (!(await u.dirHasContent(dirpath))) {
+    const message = [
+      explainMissingContent,
+      `    ${color.cyan(dirpath)}`,
+      "but no content was found.\n",
+      "Would you like to clone the content from:",
+      `    ${color.cyan(repoUrl)}?`
```

Comment:
The old `start_course.sh` script would clone repos under certain circumstances. (I removed that behavior in #964). However, it did not prompt for confirmation.

I really like prompting for confirmation in this case. Not so much to ask for permission, but to pause the flood of stdoud and alert the user "hey, we're creating this new thing at location. XYZ and it's kind of important".